io/disk/disk_info.py

```diff
@@ -0,0 +1,242 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2017 IBM
+# Author: Pridhiviraj Paidipeddi <ppaidipe@linux.vnet.ibm.com>
+
+"""
+Disk Info tests various storage block device list tools, and also it creates
+filesystems on test disk and mount it on OS boot disk where the test code
+available. Then it verifies all the tools with certain parameters like disk
+name, Size, UUID, mount points and IO Sector sizes
+"""
+
+import os
+import time
+import re
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+from avocado.utils import genio
+from avocado.utils import process, distro
+from avocado.utils.partition import Partition
+from avocado.utils.software_manager import SoftwareManager
+
+
+class DiskInfo(Test):
+
+    """
+    DiskInfo test for different storage block device tools
+    """
+
+    def setUp(self):
+        """
+        Verifies if we have list of packages installed on OS
+        and also skips the test if user gives the current OS boot disk as disk input
+        it may erase the data
+        :param disk: test disk where the disk operations can be done
+        :param fs: type of filesystem to create
+        :param dir: path of the directory to mount the disk device
+        """
+        smm = SoftwareManager()
+        pkg = ""
+        cmd = "uname -p"
+        if 'ppc' not in process.system_output(cmd, ignore_status=True):
+            self.skip("Processor is not ppc64")
+        self.disk = self.params.get('disk', default=None)
+        self.dir = self.params.get('dir', default=self.srcdir)
+        self.fstype = self.params.get('fs', default='ext4')
+        self.log.info("disk: %s, dir: %s, fstype: %s"
+                      % (self.disk, self.dir, self.fstype))
+        cmd = "df --output=source"
+        if self.disk in process.system_output(cmd, ignore_status=True):
+            self.skip("Given disk is os boot disk,"
+                      "it will be harmful to run this test")
+        if self.fstype == 'xfs':
```

Comment:
xfs is not given as fstype from yaml ? or may be you need to add a check for ext4 ?